cmd/backend/main.go

```diff
@@ -36,34 +36,21 @@ func main() {
 	log.SetOutput(f)
 	log.SetFormatter(&log.JSONFormatter{})
 
-	// can have frontend parameters come go straight into simEnv
-	foodOnPlatform := food.FoodType(100)
-	numOfAgents := []int{2, 2, 2, 2, 2, 2, 2} //agent1, agent2, team3, team4EvoAgent, team 5, team6, randomAgent
-	agentHP := 100
-	agentsPerFloor := 1 //more than one not currently supported
-	numberOfFloors := simulation.Sum(numOfAgents) / agentsPerFloor
-	ticksPerFloor := 10
+	//processing the command-line flags
+	//currently only one used is -configpath, but we are likely to need more in the near future.
+	// if not set, it uses default of "config.json"
+	configPathPtr := flag.String("configpath", "config.json", "path for parameter configuration json file")
 
-	ticksPerDay := numberOfFloors * ticksPerFloor
-	simDays := 8
-	reshuffleDays := 1
-	dayInfo := day.NewDayInfo(ticksPerFloor, ticksPerDay, simDays, reshuffleDays)
-
-	// define health parameters
-	maxHP := 100
-	weakLevel := 10
-	width := 45.0
-	tau := 10.0
-	hpReqCToW := 2
-	hpCritical := 5
-	maxDayCritical := 3
-	HPLossBase := 10
-	HPLossSlope := 0.25
+	parameters, err := config.LoadParamFromJson(*configPathPtr)
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
 
-	healthInfo := health.NewHealthInfo(maxHP, weakLevel, width, tau, hpReqCToW, hpCritical, maxDayCritical, HPLossBase, HPLossSlope)
+	healthInfo := health.NewHealthInfo(parameters.MaxHP, parameters.WeakLevel, parameters.Width, parameters.Tau, parameters.HpReqCToW, parameters.HpCritical, parameters.MaxDayCritical, parameters.HPLossBase, parameters.HPLossSlope)
 
 	// TODO: agentParameters - struct
 
-	simEnv := simulation.NewSimEnv(foodOnPlatform, numOfAgents, agentHP, agentsPerFloor, dayInfo, healthInfo)
+	simEnv := simulation.NewSimEnv(parameters.FoodOnPlatform, parameters.NumOfAgents, parameters.AgentHP, parameters.AgentsPerFloor, parameters.DayInfo, healthInfo)
```

Comment:
I would argue it's worth passing in a pointer to `parameters` to both `simulation.NewSimEnv` and `health.NewHealthInfo` and setting the parameters there. Makes it more readable imo.

Reply:
I've implemented this now, thanks.